src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -32,8 +32,8 @@ internal sealed partial class RemoteWorkspace : Workspace
         private int _currentRemoteWorkspaceVersion = -1;
 
         // internal for testing purposes.
-        internal RemoteWorkspace(HostServices hostServices, string? workspaceKind)
-            : base(hostServices, workspaceKind)
+        internal RemoteWorkspace(HostServices hostServices)
+            : base(hostServices, WorkspaceKind.RemoteWorkspace)
```

Comment:
>WorkspaceKind [](http://example.com/codeflow?start=33&length=13)

Is the same RemoteWorkspace instance gonna be used for all remote workspaces? Now that we have WorkspaceKind in Solution snapshot would it be different from Solution.Workspace.Kind for RemoteWorkspace?

Reply:
ideally, RemoteWorkspace will go away.  Since we don't want anything on OOP to even be able to go from solution->Workspace anymore, and there's no way to mutate things in OOP.  

This is one of those PRs in the "help cyrus poke around this space to learn what is going on and what unknown unknowns exist" space.

Reply:
Would we return null from Solution.Workspace then?

Reply:
that, or just throwing.  i'm experimenting with that in another branch right now.  we would at least 'ban' it. :)